app/components/sidebars/main/channels_list/list/list.js

```diff
@@ -39,15 +39,17 @@ let UnreadIndicator = null;
 export default class List extends PureComponent {
     static propTypes = {
         testID: PropTypes.string,
+        styles: PropTypes.object.isRequired,
+        theme: PropTypes.object.isRequired,
+        onSelectChannel: PropTypes.func.isRequired,
         canJoinPublicChannels: PropTypes.bool.isRequired,
         canCreatePrivateChannels: PropTypes.bool.isRequired,
         canCreatePublicChannels: PropTypes.bool.isRequired,
-        favoriteChannelIds: PropTypes.array.isRequired,
-        onSelectChannel: PropTypes.func.isRequired,
+        showLegacySidebar: PropTypes.bool.isRequired,
```

Comment:
should not be required, in fact perhaps it should not be here at all and use the fact that `channelsByCategories` is not set to show the legacy sidebar